clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -775,16 +791,40 @@ def write_bids(
             metadata["bids_filename"],
             True,
         )
-        if (
-            "dwi" in metadata["bids_filename"]
-            and "Philips" in metadata.manufacturer
-            and dcm2niix_success
-        ):
-            merge_philips_diffusion(
-                to / Path(bids_full_path).with_suffix(".json"),
-                metadata.number_of_parts,
-                metadata.run_num,
+        if dcm2niix_success:
+            scans_filepath = (
+                to
+                / str(metadata.participant_id)
+                / str(metadata.session_id)
+                / f"{metadata.participant_id}_{metadata.session_id}_scan.tsv"
+            )
+            if not Path(scans_filepath).exists():
+                index_to_write = scans_columns_names.to_csv(
+                    index=False, header=False, lineterminator="\t"
+                )
```

Comment:
I'm not sure to understand why you need to hardcode the column names and do all these steps. You basically want to write the column names (i.e. the header) only once, when creating the file. So something like this should be enough, right ?

```python
if dcm2niix_success:
    scans_filepath = ...
    row_to_write = metadata.drop(["participant_id", "session_id"]).to_csv(
        index=False,
        header=not scans_filepath.exists(),
        lineterminator="\t",
    ).rstrip("\t")
with open(scans_filepath, "a") as scans_file:
    scans_file.write(f"{row_to_write}\n")
```

Reply:
Hmmm, indeed it's not working as expected. Sorry about that, writing code in chat boxes without testing it is not without danger...

I had a quick look and this is the most readable way to do what you want that I could find (should work this time but to be tested in your settings...):

```python
if dcm2niix_success:
    scans_filepath = ...
    row_to_write = _serialize_row(
        metadata.drop(["participant_id", "session_id"]),
        write_column_names=not scan_filepath.exists(),
    )
with open(scans_filepath, "a") as scans_file:
    scans_file.write(f"{row_to_write}\n")


def _serialize_row(row: pd.Series, write_column_names: bool) -> str:
    row_dict = row.to_dict()
    to_write = [row_dict.keys(), row_dict.values()] if write_column_names else [row_dict.values()]
    return "\n".join([serialize_list(list(_)) for _ in to_write])


def _serialize_list(data: list, sep="\t") -> str:
    return sep.join([str(value) for value in data])
```

Reply:
That works and it is definitely a whole lot better than what i've done